bait-files/src/test/java/com/google/sps/UserTest.java

```diff
@@ -0,0 +1,131 @@
+package com.google.sps;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static com.google.sps.data.Keys.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.After;
+import org.junit.Test;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import java.io.*;
+import javax.servlet.http.*;
+import javax.servlet.ServletException;
+import com.google.sps.servlets.UsernameTakenServlet;
+import com.google.sps.servlets.LoginServlet;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class UserTest {
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true).setEnvIsLoggedIn(true).setEnvEmail("andrew").setEnvAuthDomain("gmail.com");;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  // Database is empty, so username should not be taken
+  @Test
+  public void emptyUsernameTakenTest() throws IOException {
+        HttpServletRequest request = mock(HttpServletRequest.class);       
+        HttpServletResponse response = mock(HttpServletResponse.class);    
+
+        when(request.getParameter("username")).thenReturn("Andrew");
+
+        StringWriter stringWriter = new StringWriter();
+        PrintWriter writer = new PrintWriter(stringWriter);
+
+        when(response.getWriter()).thenReturn(writer);
+        new UsernameTakenServlet().doGet(request, response);
+
+        verify(request, atLeast(1)).getParameter("username"); // only if you want to verify username was called...
+        writer.flush(); // it may not have been flushed yet...
+        assertTrue(stringWriter.toString().contains("false"));
+  }
+
+  // Username field is null, so return that the username is taken
+  @Test
+  public void nullUsernameTest() throws IOException {
```

Comment:
Same comment about test naming.. maybe doGet_nullRequestedUsername_returnsTaken.

As a final note, the word "returns" in the last part is kind of optional since with this three part naming style, it can be kind of understood that the last part means what is returned.

Reply:
renamed